strawberryfields/api_client.py

```diff
@@ -0,0 +1,584 @@
+# Copyright 2019 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+APIClient library
+================
+
+**Module name:** :mod:`strawberryfields.api_client`
+
+.. currentmodule:: strawberryfields.api_client
+
+
+This module provides a thin client that communicates with the Xanadu Platform API over the HTTP
+protocol, based on the requests module. It also provides helper classes to facilitate interacting
+with this API via the Resource subclasses, as well as the ResourceManager wrapper around APIClient
+that is available for each resource.
+
+A single APIClient instance can be used throughout one's session in the application. The application
+will attempt to configure the APIClient instance using a configuration file or defaults, but the
+user can choose to override various parameters of the APIClient manually.
+
+A typical use looks like this:
+    .. code-block:: python
+
+        job = Job()
+        circuit = '''
+            name StateTeleportation
+            version 1.0
+            target gaussian (shots=1000)
+
+            complex alpha = 1+0.5j
+            Coherent(alpha) | 0
+            Squeezed(-4) | 1
+            Squeezed(4) | 2
+            BSgate(pi/4, 0) | (1, 2)
+            BSgate(pi/4, 0) | (0, 1)
+            MeasureX | 0
+            MeasureP | 1
+            Xgate(sqrt(2)*q0) | 2
+            Zgate(sqrt(2)*q1) | 2
+            MeasureHeterodyne() | 2
+        '''
+        job.manager.create(circuit=circuit)
+
+        job.id  # Returns the job ID that was generated by the server
+        job.reload()  # Fetches the latest job data from the server
+        job.status  # Prints the status of this job
+        job.result  # Returns a JobResult object
+        job.circuit  # Returns a JobCircuit object
```

Comment:
API looks awesome!

Reply:
There's a slight fix to this doc that I will push shortly. There are also a few changes to the API that I'd like to make in a follow up PR to simplify the API a little.